src/jvm/test/org/jetbrains/kotlinx/lincheck_test/gpmc/coroutines/DaemonThreadPoolTest.kt

```diff
@@ -11,24 +11,158 @@
 package org.jetbrains.kotlinx.lincheck_test.gpmc.coroutines
 
 import kotlinx.coroutines.asCoroutineDispatcher
+import kotlinx.coroutines.launch
 import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.sync.Mutex
 import org.jetbrains.kotlinx.lincheck.ExperimentalModelCheckingAPI
+import org.jetbrains.kotlinx.lincheck.LincheckAssertionError
+import org.jetbrains.kotlinx.lincheck.isInTraceDebuggerMode
 import org.jetbrains.kotlinx.lincheck.runConcurrentTest
-import org.junit.Ignore
+import org.junit.Assume.assumeFalse
+import org.junit.Before
 import org.junit.Test
 import java.util.concurrent.Executors
+import java.util.concurrent.atomic.AtomicInteger
+import kotlin.concurrent.thread
 
 @OptIn(ExperimentalModelCheckingAPI::class)
 class DaemonThreadPoolTest {
 
-    @Ignore("Daemon threads are treated as active, see https://github.com/JetBrains/lincheck/issues/542")
+    @Before // spin-loop detection is unsupported in trace debugger mode
+    fun setUp() = assumeFalse(isInTraceDebuggerMode)
+
     @Test
     fun testUselessThreadPool() {
         runConcurrentTest(1000) {
             val pool = Executors.newFixedThreadPool(1).asCoroutineDispatcher()
             runBlocking(pool) { /* do nothing */ }
-            // pool.close() -- this call tells gpmc that threads are not in infinite spinning,
-            //                 but interleavings exploration still happens
+            // pool.close() -- we do not need this call, because lincheck will
+            //                 abort background threads, when main is finished
+        }
+    }
+
+    @Test
+    fun testUselessThreadPoolWithMultipleCoroutines() {
```

Comment:
Would make sense to rename the test to something like `testNonClosedThreadPoolCoroutineDispatcher`